BTrDB.cs

```diff
@@ -1,131 +1,732 @@
+        /// </summary>
+        public void Reset()
+        {
+            throw new InvalidOperationException("BTrDB cursors can not be reset");
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+    }
+
+    
+    /// <summary>
+    /// Stream represents a stream in BTrDB, it is the primary way of inserting and querying data
+    /// </summary>
     public class Stream
     {
-        private Guid id;
-        private BTrDB db;
-        internal Stream(Guid id, BTrDB db)
+        /// <summary>
+        /// Handle to the GRPC conn
+        /// </summary>
+        private readonly BTrDB db;
+
+        /// <summary>
+        /// The UUID of the stream
+        /// </summary>
+        private readonly Guid id;
+       
+        /// <summary>
+        /// The collection. May be null
+        /// </summary>
+        private string collection;
+
+        /// <summary>
+        /// The tags for the stream, may be null
+        /// </summary>
+        private Dictionary<string, string> tags;
+
+        /// <summary>
+        /// The annotations for the stream, may be null
+        /// </summary>
+        private Dictionary<string, string> annotations;
+
+        /// <summary>
+        /// The property version associated with the last refresh, may be zero
+        /// </summary>
+        private ulong propertyVersion;
+
+        /// <summary>
+        /// Create a stream with just it's uuid
+        /// </summary>
+        internal Stream( BTrDB db, Guid id)
+        {
+            this.id = id;
+            this.db = db;
+        }
+
+        /// <summary>
+        /// Create a stream, specifying all it's information
+        /// </summary>
+        internal Stream(BTrDB db, Guid id, string collection, Dictionary<string, string> tags, Dictionary<string, string> annotations)
         {
             this.id = id;
             this.db = db;
+            this.tags = tags;
+            this.annotations = annotations;
+        }
+
+        /// <summary>
+        /// Create a stream from the protobuf descriptor
+        /// </summary>
+        internal Stream(BTrDB db, V5Api.StreamDescriptor desc)
+        {
+            this.db = db;
+            id = new Guid(desc.Uuid.ToByteArray());
+            collection = desc.Collection;
+            tags = desc.Tags.ToDictionary(e => e.Key, e => e.Val?.Value);
+            annotations = desc.Annotations.ToDictionary(e => e.Key, e => e.Val?.Value);
+            propertyVersion = desc.PropertyVersion;
         }
 
+        /// <summary>
+        /// Check if the stream exists
+        /// </summary>
+        /// <returns>true if the stream exists, false otherwise</returns>
         public bool Exists()
         {
-            throw new NotImplementedException("not implemented");
+            var p = new V5Api.StreamInfoParams
+            {
+                Uuid = ByteString.CopyFrom(id.ToByteArray()),
```

Comment:
This is a go programmer thing to caps the Uuid? Or is this also a C# thing?

Reply:
This is actually a property, so it should be capitalized as part of C# style. More specifically though, this property is autogenerated from protobuf